### DISCUSSION TITLE
Formula field not syncing with Firestore

### DISCUSSION THREAD

**Discussion by darkmayuk** [AntlerVC/firetable#1298] [Support / Q&A] [answered]
Again, as per my other question, I am a new user of Rowy, so maybe I'm missing something obvious. But I am running into the following issue:

I create a calculated field, type Formula. It is a simple formula which concatenates a fixed string with a string read from a different field in the same row. It works fine within Rowy. But the problem is that this field does not propogate to Firestore. When I check in Firestore all the other fields have propogated successfully, but not this calculated (Formula) field. It simply does not exist in the corresponding Firestore documents.

**Comment by BeeBombshell** [comment, marked as answer]
Hey @darkmayuk, Formula fields in the database are **only intended for UI purposes** and are not permanently saved. If you require the data to be stored in Firestore, we highly recommend using the [Derivative Field](https://docs.rowy.io/field-types/derivative) instead. The Derivative Field operates similarly to a Formula Field but offers more advanced functionality. It has the ability to persist data to Firestore in real-time based on data triggers within the table. This offers a powerful solution for storing and managing data effectively in Firestore.

**Reply by darkmayuk** [reply]
OK, thanks - I will try out Derivative fields instead for this purpose.
